test/functional/test_framework/util.py

```diff
@@ -806,3 +806,6 @@ def make_vote_tx(finalizer, finalizer_address, target_hash, source_epoch, target
     vtx = finalizer.createvotetransaction(vote, input_tx_id)
     vtx = finalizer.signrawtransaction(vtx)
     return vtx['hex']
+
+def generate_block(proposer, count=1):
+    return proposer.generatetoaddress(count, proposer.getnewaddress('', 'bech32'))
```

Comment:
nit: technically it can be any node that will generate the block so I'd call it `node` instead of `proposer`

Reply:
The question. Although at the moment finalizer can generate blocks, we do not do that, and logically only proposer should generate. Could you maybe elaborate a bit on your opinion?

Reply:
We can start the node with the flag `-proposing=0`. then this node is technically not a proposer but it still creates blocks. 
And also the point about `finalizer` that you said, this function would even accept it and will create a block. 
Since there is no restriction which type of the node we accept, I thought that the generic name fits better here.

Reply:
Not sure I agree about using the generic name here, even if theoretically we can generate blocks using any node.

Reply:
>  logically only proposer should generate

any one can generate, including finalizers; you can be finalizer and proposer at the same time. There's no reason to forbid that, as someone operating a finalizer can just spin up the software a second time with same keys and operate as a proposer too.

Obviously you can't spend funds which are locked by having made a deposit, so you can't use them to propose (please correct me if I'm wrong), so if all your stake is on that deposit then true, you can't propose.

@Gnappuraz Right?

Reply:
Renamed it anyway.